hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionScannerImpl.java

```diff
@@ -629,17 +640,37 @@ private boolean nextInternal(List<Cell> results, ScannerContext scannerContext)
     }
   }
 
-  private void incrementCountOfRowsFilteredMetric(ScannerContext scannerContext) {
+  private boolean incrementCountOfRowsFilteredMetric(ScannerContext scannerContext)
+    throws DoNotRetryIOException {
     region.filteredReadRequestsCount.increment();
     if (region.getMetrics() != null) {
       region.getMetrics().updateFilteredRecords();
     }
 
     if (scannerContext == null || !scannerContext.isTrackingMetrics()) {
-      return;
+      return false;
+    }
+
+    long countOfRowsFiltered = scannerContext.getMetrics().countOfRowsFiltered.incrementAndGet();
```

Comment:
We'd better not mix things up with metrics? Metrics is not a critical thing usually.